main.go

```diff
@@ -197,3 +204,11 @@ func readStdin() (string, error) {
 		return "", nil
 	}
 }
+
+func versionInfo() string {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return "(devel)"
+	}
+	return info.Main.Version
```

Comment:
Does this work? It looks like `info.Main.Version` still always returns `(devel)`...? https://groups.google.com/g/golang-nuts/c/ObM-ilI7sio

Reply:
Thanks for reviewing.🙏

This way is used by [yo](https://github.com/cloudspannerecosystem/yo/blob/44c1d636a3cb215cbf4e45893496770b536b032e/v2/cmd/root.go#L71) and [wrench](https://github.com/cloudspannerecosystem/wrench/blob/9432313f606f9410528612e935ffb3f2f9b6d42c/cmd/root.go#L108) and they return their version properly.

It seems that it shows the valid version when it is released or installed via `go install` but I don't understand its mechanism. I'll look into it later.

Reply:
FYI: I believe it should be only valid in `go install` with tags.
Released binary should be use another mechanism like `LDFLAGS`( spanner-cli don't have release binaries).

I think you all are Japanese
https://rhysd.hatenablog.com/entry/2021/06/27/222254

Reply:
Oh I didn't know that. Then maybe we can merge this PR and see how it goes.

Reply:
It looks like pkg.go.dev correctly recognizes the version of spanner-cli from git tag: https://pkg.go.dev/github.com/cloudspannerecosystem/spanner-cli

> FYI: I believe it should be only valid in go install with tags.

Do you mean `go install github.com/cloudspannerecosystem/spanner-cli@latest` always shows `latest`? Or does it show the git tag associated with the latest git commit?

Reply:
@apstndb 
Thanks for following up!

It's exactly what @apstndb  said, but I investigated real examples too. So I leave it.

According to https://github.com/golang/go/issues/50603, `info.Main.Version` return different result when the binary was built via `go build` and when it was built via `go install`.

> The version information, in the sense of Go's pseudo version is not recorded for the main module when doing go build:
> 
> ```
> : emerald:ver; go build
> : emerald:ver; go version -m hello | grep 'mod.*hello'
> 	mod	mgk.ro/hello	(devel)	
> : emerald:ver; 
> ```
> The version is recorded as expected when doing go install:
> 
> ```
> : emerald:ver; go install robpike.io/ivy@latest
> go: downloading robpike.io/ivy v0.1.124
> : emerald:ver; go version -m `which ivy` | grep 'mod.*ivy'
> 	mod	robpike.io/ivy	v0.1.12	h1:qI7dnEiXhorB+za07W6qX3sG+IvBK4EUl38vUHAf53Q=
> : emerald:ver; 
> : emerald:ver; 
> ```


As a workaround for this:

1. [Define a `version` variable](https://github.com/cloudspannerecosystem/yo/blob/44c1d636a3cb215cbf4e45893496770b536b032e/v2/cmd/root.go#L40) in go program.
2. [Use `ldflags` to inject its version into `version`](https://github.com/cloudspannerecosystem/yo/blob/44c1d636a3cb215cbf4e45893496770b536b032e/.goreleaser.yml#L19).

Reply:
Thank you all for the information! The official installation of spanner-cli is `go install` as mentioned in README, so it looks like we don't need to manually embed the version information.

If it's fine, I'll go ahead and merge this PR.

Reply:
> Do you mean go install github.com/cloudspannerecosystem/spanner-cli@latest always shows latest? Or does it show the git tag associated with the latest git commit?

Oh, that was not precise answer.
It always shows module version when `go install`-ed, so it can print version even if not tagged.

```
$ go install github.com/cloudspannerecosystem/wrench@9432313  
go: downloading github.com/cloudspannerecosystem/wrench v1.10.2-0.20241120031843-9432313f606f
$ wrench --version
v1.10.2-0.20241120031843-9432313f606f
```

Reply:
Thank you, sounds good.

Reply:
I merged the PR and put the tag `v0.10.9`: https://github.com/cloudspannerecosystem/spanner-cli/releases/tag/v0.10.9

Reply:
Thank you!

<img width="266" alt="image" src="https://github.com/user-attachments/assets/23102b0d-1672-4f6b-a3aa-6c7554616a89">